### PR TITLE
Fix CUDA errors in submdspan about int to size_t narrowing

### PR DESCRIPTION
**Pull request by aprokop**
I was getting errors on CUDA 11.5.2 with GCC 9.3.0:
```
<snip>/kokkos/core/src/../../tpls/mdspan/include/mdspan/../experimental/__p2630_bits/submdspan_mapping.hpp(134): error: invalid narrowing conversion from "int" to "size_t"
<snip>/kokkos/core/src/../../tpls/mdspan/include/mdspan/../experimental/__p2630_bits/submdspan_mapping.hpp(364): error: invalid narrowing conversion from "int" to "size_t"
```
when building Kokkos `develop` (24454fa82).

I don't really know what I'm doing, so feel free to edit however you need. This fixes it for this particular build, but probably not the best way to go.

### CODE REVIEW

File: include/experimental/__p2630_bits/submdspan_mapping.hpp [hunk 1]
```diff
@@ -131,7 +131,7 @@ struct deduce_layout_left_submapping<
     IndexType, SubRank, std::index_sequence<Idx...>, SliceSpecifiers...> {
 
   using count_range = index_sequence_scan_impl<
-      0, (is_index_slice_v<SliceSpecifiers, IndexType> ? 0 : 1)...>;
+      0, (is_index_slice_v<SliceSpecifiers, IndexType> ? (size_t)0 : (size_t)1)...>;
```
Comment (masterleinad) [inline include/experimental/__p2630_bits/submdspan_mapping.hpp]:
```suggestion
      0u, (is_index_slice_v<SliceSpecifiers, IndexType> ? 0u : 1u)...>;
```

File: include/experimental/__p2630_bits/submdspan_mapping.hpp [hunk 2]
```diff
@@ -361,7 +361,7 @@ struct deduce_layout_right_submapping<
 
   static constexpr size_t Rank = sizeof...(Idx);
   using count_range = index_sequence_scan_impl<
-      0, (std::is_convertible_v<SliceSpecifiers, IndexType> ? 0 : 1)...>;
+      0, (std::is_convertible_v<SliceSpecifiers, IndexType> ? (size_t)0 : (size_t)1)...>;
```
Comment (masterleinad) [inline include/experimental/__p2630_bits/submdspan_mapping.hpp]:
```suggestion
      0u, (std::is_convertible_v<SliceSpecifiers, IndexType> ? 0u : 1u)...>;
```

**Review by mhoemmen — APPROVED** [review]
LWG (in WG21) has expressed preference for integer literals (e.g., `0u` or even `0UZ` for a `size_t` literal) over casts.  It's not a big deal, though.

**Review by masterleinad — APPROVED** [review]

**Comment by ndellingwood** [comment]
@crtrott can this be merged and patched into Kokkos? Lots of warnings showing up in Cuda builds

**Pull request merged**